Add Bitcoin node version informatino

laanwj opened the pull request:
This does the bare minimum to communicate with a Bitcoin node and gets its version message. Add the user agent and protocol version to the report.

This is the first Go code I've ever written, so it's probably a mess, sorry for that.

Example:
```
./onionscan -verbose nns4r54x3lfbrkq5.onion                                                  
2016/09/26 23:02:17 Starting Scan of nns4r54x3lfbrkq5.onion
2016/09/26 23:02:17 This might take a few minutes..
...
2016/09/26 23:02:27 INFO: Checking nns4r54x3lfbrkq5.onion Bitcoin(8333)
2016/09/26 23:02:28 INFO: Detected possible Bitcoin instance
2016/09/26 23:02:29 INFO: Found Bitcoin version: /Satoshi:0.13.99(Ereshkigal)/ (70014)
```

s-rah commented:
This is awesome! Thanks for the PR - I'll take a look at code and get this merged in soon.

laanwj commented:
Cool, let me know if anything should be done differently.

A future idea might be to use `getaddr`/`addr` messages to find other .onion peers to add to the scan.

s-rah commented on protocol/bitcoin_scanner.go:
```diff
@@ -10,6 +16,130 @@ import (
+// Message start of packets on mainnet
+var MsgStartMainnet = []byte {0xf9,0xbe,0xb4,0xd9}
+// User agent to send to scanned nodes
+const user_agent = "/OnionScanner:0.0.1/"
```
User agent should be OnionScan

s-rah commented on protocol/bitcoin_scanner.go:
```diff
@@ -2,6 +2,12 @@ package protocol
 
```
Please run gofmt on these files.

s-rah commented on protocol/bitcoin_scanner.go:
```diff
@@ -10,6 +16,130 @@ import (
+	 // Only one-byte CompactSizes for now, this string can be 256 bytes max anyway and it's hardly ever longer than 100
+	if user_agent_length < 253 {
+		report.BitcoinUserAgent = string(payload[81:81+user_agent_length])
+	}
```
add an error condition here.

laanwj commented:
Done! Pushed a new version
- Changed user agent to OnionScan instead of OnionScanner
- Added error condition for >253 user agent case

And ran the source file through gofmt (without any specific arguments).

s-rah commented:
Thanks!
